AdyenDropIn/Components/PreApplePayComponent.swift

```diff
@@ -65,27 +59,30 @@ internal final class PreApplePayComponent: PresentableComponent, FinalizableComp
     /// :nodoc:
     internal init(paymentMethod: ApplePayPaymentMethod,
                   apiContext: APIContext,
-                  payment: Payment,
                   configuration: Configuration,
                   applePayConfiguration: ApplePayComponent.Configuration) throws {
         self.apiContext = apiContext
-        self._payment = payment
         self.paymentMethod = paymentMethod
         self.configuration = configuration
-
+        self.payment = applePayConfiguration.applePayPayment.payment
         self.applePayComponent = try ApplePayComponent(paymentMethod: paymentMethod,
                                                        apiContext: apiContext,
-                                                       payment: payment,
                                                        configuration: applePayConfiguration)
         self.applePayComponent.delegate = self
     }
 
-    /// :nodoc:
+    internal func didCancel() {
+        let rootViewController = UIApplication.shared.keyWindow?.rootViewController
+        if let navigation = presentationDelegate,
```

Comment:
Why is having the `presentationDelegate` not enough and we have to check for the `rootViewController`?
Could be nice to add this as a comment in the code

Reply:
I have changed it a bit - it should be more clear now :)